prov/verbs/src/verbs_eq.c

```diff
@@ -766,6 +833,24 @@ fi_ibv_eq_read(struct fid_eq *eq_fid, uint32_t *event,
 	if ((ret = fi_ibv_eq_read_event(eq, event, buf, len, flags)))
 		return ret;
 
+	if (domain) {
+		/* Non-blocking call to retrieve IB async events */
+		if (fi_ibv_read_async_event(domain, eq)) {
```

Comment:
Need to hold `eq->lock` before calling this function as it may access `eq->err` or setting `eq->err` can be pull out of the function to keep it simple.